activesupport/lib/active_support/testing/event_reporter_assertions.rb

```diff
@@ -0,0 +1,164 @@
+# typed: true
+# frozen_string_literal: true
+
+module ActiveSupport
+  module Testing
+    # Provides test helpers for asserting on ActiveSupport::EventReporter events.
+    module EventReporterAssertions
+      module EventCollector # :nodoc:
+        @subscribed = false
+        @mutex = Mutex.new
+
+        class Event # :nodoc:
+          attr_reader :event_data
+
+          def initialize(event_data)
+            @event_data = event_data
+          end
+
+          def inspect
+            "#{event_data[:name]} (payload: #{event_data[:payload].inspect}, tags: #{event_data[:tags].inspect})"
+          end
+
+          def matches?(name, payload, tags)
+            return false unless name == event_data[:name]
```

Comment:
```suggestion
            return false unless name.to_s == event_data[:name]
```
We talked about this internally but currently, event names when sent are standardized to strings, this results in cases like

```
assert_event_reported(:user_created, payload: { id: 123 }) do
  Rails.event.notify(:user_created, { id: 123 })
end
```
not intuitively working correctly, as `"user_created"` != :user_created.

If the event name also standardizes on symbols like the payloads this will need to correspondingly be changed to ensure that reported events & the assertion are normalized on the same type (`symbol` & `symbol` or `string` & `string`)

Reply:
I think standardizing on strings for naming makes sense, but agree that the helper should be more flexible. Applied your suggestion 👍